### DISCUSSION TITLE
Yarn 2 support?

### DISCUSSION THREAD

**Discussion by FezVrasta** [vercel/community#551] [Help] [answered]
I was reading Yarn 2 is not supported for some deployment kinds:

https://vercel.com/support/articles/does-vercel-support-yarn-2

I was wondering what's the reason? Does the above applies even if we are using the Yarn 3.x binaries directly committed into the repository and automatically loaded by Yarn 1.x?

**Comment by samcx** [comment, marked as answer]
No reason! Full support v2/3 is coming 👀

**Reply by Tigatok** [reply]
@samsisle awesome! Any ETA or roadmap for Yarn 3?

**Reply by FezVrasta** [reply]
For what is worth, it seems to already work for us, Yarn 1.x supports the ability to auto-upgrade to any later version if the binary is committed in the repository.

**Reply by Tigatok** [reply]
@FezVrasta are you saying yarn 3 works for you in the pipeline?

**Reply by FezVrasta** [reply]
Yes it does

**Reply by Tigatok** [reply]
@FezVrasta Interesting, when I try to make a build the new .yarn and .yarnrc.yml prevent it from running due to the version not existing in the pipeline. If I remove those, it just runs 1.22.x How did you get it to run using 3 in the deployment?

**Reply by FezVrasta** [reply]
I have this in `.yarnrc.yml`

```
yarnPath: .yarn/releases/yarn-3.2.0.cjs
```

and then I have the `.cjs` bundle/binary in that path.

**Reply by samcx** [reply]
@Tigatok No ETA unfortunately!

**Reply by dannyrb** [reply]
FWIW, we are also using Yarn 3 successfully for some projects.
